vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/StaffSearchManagerTest.java

```diff
@@ -0,0 +1,344 @@
+            spaces.add(space);
+        }
+
+        modules = new ArrayList<>();
+        StringBuilder moduleIdBuilder = new StringBuilder();
+        for (int i = 1; i <= 10; i++) {
+            Module module = new Module();
+            moduleIdBuilder.setLength(0);
+            moduleIdBuilder.append("MODULE_ID_");
+            moduleIdBuilder.append(String.valueOf(i));
+            module.setId(moduleIdBuilder.toString());
+            modules.add(module);
+        }
+
+        slides = new ArrayList<>();
+        StringBuilder slideIdBuilder = new StringBuilder();
+        for (int i = 1; i <= 10; i++) {
+            Slide slide = new Slide();
+            slideIdBuilder.setLength(0);
+            slideIdBuilder.append("SLIDE_ID_");
+            slideIdBuilder.append(String.valueOf(i));
+            slide.setId(slideIdBuilder.toString());
+            slides.add(slide);
+        }
+
+        slideTexts = new ArrayList<>();
+        StringBuilder slideTextIdBuilder = new StringBuilder();
+        for (int i = 1; i <= 10; i++) {
+            Slide slideText = new Slide();
+            slideTextIdBuilder.setLength(0);
+            slideTextIdBuilder.append("SLIDETEXT_ID_");
+            slideTextIdBuilder.append(String.valueOf(i));
+            slideText.setId(slideTextIdBuilder.toString());
+            slideTexts.add(slideText);
+        }
+
+        ReflectionTestUtils.setField(serviceToTest, "pageSize", 10);
+    }
+
+    @Test
+    public void test_searchInSpaces_success() {
+
+        Pageable requestedPage = PageRequest.of(0, 10);
+        String search = "space";
+        when(spaceManager.findByNameOrDescription(requestedPage, search))
+                .thenReturn(new PageImpl<Space>(spaces,requestedPage, 12));
+        Page<Space> tempResult = serviceToTest.searchInSpaces(search, 1);
+        assertEquals(10, tempResult.getContent().size());
+        List<String> idList = tempResult.stream().map(space -> space.getId()).collect(Collectors.toList());
+        StringBuilder dummySpaceIdBuilder = new StringBuilder();
+        for (int i = 1; i <= 10; i++) {
+            dummySpaceIdBuilder.setLength(0);
+            dummySpaceIdBuilder.append("SPACE_ID_");
+            dummySpaceIdBuilder.append(String.valueOf(i));
+            assertEquals(dummySpaceIdBuilder.toString(),idList.get(i-1));
+        }
+        verify(spaceManager).findByNameOrDescription(requestedPage, search);
+    }
+
+    @Test
+    public void test_searchInSpaces_pageGreaterThanTotalPages() {
+
+        Pageable requestedPage = PageRequest.of(6, 10);
+        String search = "space";
+        when(spaceManager.findByNameOrDescription(requestedPage, search)).thenReturn(new PageImpl<Space>(new ArrayList<Space>(),requestedPage, 12));
+        Pageable requestedPage1 = PageRequest.of(1, 10);
+        when(spaceManager.findByNameOrDescription(requestedPage1, search)).thenReturn(new PageImpl<Space>(spaces,requestedPage1, 12));
```

Comment:
this logic seems wrong or am I missing something?
the requested page is 7 with 10 items. there are a total of 12 items. the search returns page 1 (it starts at 0, right?) with 10 items, but if there are 12 items total, it should only. have 2 items on it, shouldn't it? The first page has 10 items, the second one 2.

Reply:
there are two 2 pages in total but the requested page is 6.so only 1 page is being returned